pkg/controller/sparkapplication/controller.go

```diff
@@ -150,6 +173,22 @@ func newSparkApplicationController(
 	return controller
 }
 
+
+func filterSparkAndPodEventsByNamespace(regex regexp.Regexp ) func(obj interface{}) bool {
+	return func(obj interface{}) bool {
+		switch sparkObj := obj.(type) {
+		case *v1beta2.SparkApplication:
+			appNamespace := sparkObj.Namespace
+			return util.IsNamespaceMatchesFilter(regex, appNamespace)
+		case *apiv1.Pod:
+			podNamespace := sparkObj.Namespace
+			return util.IsNamespaceMatchesFilter(regex, podNamespace)
+		default:
```

Comment:
you only need to get Namespace from 'ObjectMeta' object. object type are already filtred by the informer (I think)